arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Sequence.kt

```diff
@@ -612,15 +612,19 @@ public fun <A, B> Sequence<Validated<A, B>>.separateValidated(): Pair<Sequence<A
   return asep to bsep
 }
 
+@Deprecated("Terminal operators on Sequence are being deprecated. Please use either one of the supported terminal operators of Sequence and opt for supported [traverse] functions")
```

Comment:
What is the replacement that a user would perform in this case? The current message is a bit cryptic. If we know the expression that would replace `sequenceeEther` then we can add it to a `ReplaceWith`.

Reply:
here a user could opt to use `toList().sequenceEither()` instead, if that sounds good to all, I can add it?

Reply:
There are several options here:

- Deprecate and change the signature to `Sequence<Either<E, A>>.sequenceEither(): Either<E, List<A>>`, which doesn't lie about the fact that it's terminal. This would be more optimized than `toList().sequenceEither()`.

- Deprecate and `ReplaceWith` `toList().sequenceEither()`.

Now that I think about it, the latter might be problematic since that cannot short-circuit an infinite `Sequence` on `Either.Left` but deprecating towards a `List` in the return type is going to be very difficult without breaking the Kotlin API (the binary can be kept in tact by using HIDDEN).